x-pack/plugins/ml/public/application/datavisualizer/file_based/components/fields_stats_grid/fields_stats_grid.tsx

```diff
@@ -0,0 +1,335 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useMemo, FC, useState } from 'react';
+import { EuiFlexGroup, EuiFlexItem, EuiNotificationBadge, EuiSpacer, EuiText } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { ML_JOB_FIELD_TYPES } from '../../../../../../common/constants/field_types';
+import { roundToDecimalPlace } from '../../../../formatters/round_to_decimal_place';
+import type { FindFileStructureResponse } from '../../../../../../common/types/file_datavisualizer';
+import { DataVisualizerDataGrid, ItemIdToExpandedRowMap } from '../../../stats_datagrid';
+import type { DataVisualizerFileBasedAppState } from '../../../../../../common/types/ml_url_generator';
+import { getFieldNames, getSupportedFieldType } from './get_field_names';
+import type { FileBasedFieldVisConfig } from '../../../stats_datagrid/types/field_vis_config';
+import { FileBasedDataVisualizerExpandedRow } from '../expanded_row';
+
+import { DataVisualizerFieldNamesFilter } from '../field_names_filter';
+import { DataVisualizerFieldTypesFilter } from '../field_types_filter';
+
+function createFields(results: FindFileStructureResponse) {
```

Comment:
`createFields` and `filterFields ` are pretty large functions and could be moved out of this file to improve readability.

Reply:
Updated here https://github.com/elastic/kibana/pull/87598/commits/09f2b31a87c750f5c0ad60a849ee9e71242b923e